app/services/charges/charge_models/percentage_service.rb

```diff
@@ -13,6 +13,35 @@ def compute_amount
         compute_percentage_amount + compute_fixed_amount
       end
 
+      def amount_details
+        paid_units = units - free_units_value
+        paid_units = 0 if paid_units.negative?
+        percentage_unit_amount = paid_units.zero? ? 0 : compute_percentage_amount.fdiv(paid_units)
+        paid_events = aggregation_result.count - free_units_count
+
+        {
+          unit_amounts: {
+            units:,
+            free_units: free_units_value,
+            paid_units:,
+            free_events: free_units_count,
+            paid_events:,
+            percentage_amount: compute_percentage_amount,
+            percentage_unit_amount:,
+            fixed_amount: compute_fixed_amount,
+            fixed_unit_amount: fixed_amount,
+            min_max_amount:,
+          }
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.